specs/sosreport_integration.adoc

```diff
@@ -0,0 +1,153 @@
+// vim: tw=79
+
+= SOS Report integration
+
+The propose of this file is to identify data-points and techniques for
+SOS Report integration with Tendrl components.
+
+== Problem description
+The current SOS report is not having any feature or plugin to analyze Tendrl components. Since the Tendrl components function as individual services, these components can be analysed with the help of SOS report in case of failures.
+
+== Use Cases
+
+One way to integrate SOS Report with Tendrl is creating plugin for each Tendrl component.
+
+* Usually SOS Report is run by an admin on each node to get the report. Assuming a multi-cluster environment with large number of nodes, two types of situations can come into picture here :
+
+** One or a few nodes fail and admin runs SOS-Report on them.
+
+** There is a multi-node failure. Will it feasible to let admin run SOS Report on all of the failed nodes?
+
+* Using policies in SOS Report it is decided how it will behave on a particular distribution. It has to be decided for which distributions the policies have to be written.
+
+== Proposed change
+
+For different Tendrl services their respective plugins have to be written.
+
+* Following are the data-points which can be used for the plugins:
+
+** Tendrl-node-agent
+*** Rpm versions for common and node-agent
+*** If tendrl-tendrl-epel-7.repo is enabled
+*** Configurations in /etc/tendrl/node-node/
+*** Status of tendrl-node-agent.socket service
+*** SELinux configurations
+*** Firewall status and configurations
+*** Package requirements
```

Comment:
One way this can be done is shipping a small script with tendrl (or as a part of the plugin itself )which can be run by the sos-report and this can be used to capture those logs.